<?xml version="1.0"?>

<launch>

	<arg name="robot_name" default="thorvald_001"/>

	<!-- gmapping -->
	<node name="gmapping" pkg="gmapping" type="slam_mapping">
            <param name="odom_frame_id" type="string" value="$(arg robot_name)/odom"/>
            <param name="base_frame_id" type="string" value="$(arg robot_name)/base_link"/>
	</node> 

	<!-- move_base -->
	<node name="move_base" pkg="move_base" type="move_base" output="screen">
        <remap from="odom" to="$(arg robot_name)/odometry/base_raw" />
        <remap from="cmd_vel" to="$(arg robot_name)/nav_vel" />

        <rosparam file="$(find bot_control)/config/costmap_common_params.yaml" command="load" ns="global_costmap"/>
        <rosparam file="$(find bot_control)/config/costmap_common_params.yaml" command="load" ns="local_costmap"/>

        <rosparam file="$(find bot_control)/config/global_costmap_params.yaml" command="load"/>
        <rosparam file="$(find bot_control)/config/local_costmap_params.yaml" command="load"/>

        <rosparam file="$(find bot_control)/config/planners.yaml" command="load"/>

	</node> 
</launch>
